server/tests/endpoints/certification/test_certification.py

```diff
@@ -94,7 +96,66 @@ def test_hardware_mismatch(generator: DataGenerator, test_client: TestClient):
                 "firmware_revision": None,
             },
             "os": {
-                "distributor": "Canonical Ltd.",
+                "distributor": "Ubuntu",
+                "version": "24.04",
+                "codename": "noble",
+                "kernel": {"name": None, "version": "5.7.1-generic", "signature": None},
+            },
+            "processor": {
+                "identifier": [],
+                "frequency": 2400,
+                "manufacturer": "AMD",
+                "version": "AMD EPYC 3251 8-Core Processor",
+            },
+        },
+    )
+
+    assert response.status_code == 200
+    assert response.json() == {"status": "Not Seen"}
+    assert "Hardware cannot be found" in caplog.text
+    assert "board model: sample board" in caplog.text
+
+
+def test_bios_mismatch(
+    caplog: LogCaptureFixture,
+    generator: DataGenerator,
+    test_client: TestClient,
+):
+    """
+    Setup where board model and vendor match but bios does not match.
+    The mismatch should be logged
+    """
+    vendor = generator.gen_vendor(name="Known Vendor")
+    board = generator.gen_device(
+        vendor,
+        identifier="dmi:0001",
+        name="FW00Q",
+        version="0001A",
+        category=DeviceCategory.BOARD,
+        bus=BusType.dmi,
+    )
+    generator.gen_bios(vendor=vendor, version="1.0.12")
+
+    response = test_client.post(
+        "/v1/certification/status",
+        json={
```

Comment:
This post is repeated across tests perhaps it's an opportunity for following the DRY principle. We could create a class with a helper method that returns some default json request prefilled with some dummy data. This function can take as input some optional parameters like vendor name and output a customized json in such situations. If we do that, then we avoid the need to repeat this structure in every test with dummy values. More importantly we decouple the structure of the request from the tests, such that if the structure changes then we only need to make a change in one place.